packages/accounts/src/decode-account.ts

```diff
@@ -44,8 +45,7 @@ export function assertAccountDecoded<TData extends object, TAddress extends stri
     account: Account<TData | Uint8Array, TAddress> | MaybeAccount<TData | Uint8Array, TAddress>,
 ): asserts account is Account<TData, TAddress> | MaybeAccount<TData, TAddress> {
     if (accountExists(account) && account.data instanceof Uint8Array) {
-        // TODO: coded error.
-        throw new Error(`Expected account [${account.address}] to be decoded.`);
+        throw new SolanaError(SOLANA_ERROR__INVALID_ACCOUNT_DATA, { addresses: [account.address] });
```

Comment:
I wonder if maybe this error and the one below should be defined as a new error code. Because technically we're not saying the account has invalid data. We're saying that, we expected a decoded account object but got an encoded one. At this point we still don't know if that encoded data is valid or not.

I'd say something like `SOLANA_ERROR__EXPECTED_DECODED_ACCOUNT` could work well.